app/server/models/transfer_account.py

```diff
@@ -194,19 +194,38 @@ def get_approval(self, receiving_address):
                 return approval
         return None
 
-    def approve_and_disburse(self):
-        if not self.is_approved:
-            self.is_approved = True
+    def approve_and_disburse(self, initial_disbursement=None, auto_resolve=False):
+        from server.utils.access_control import AccessControl
 
-            if self.is_beneficiary:
-                disbursement = self.make_initial_disbursement()
+        auto_approve = current_app.config['AUTO_APPROVE_TRANSFER_ACCOUNTS']
+        admin = getattr(g, 'user', None)
+
+        if not self.is_approved and admin:
+            if AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'superadmin'):
+                auto_resolve = True
+                self.is_approved = True
```

Comment:
is there a reason this isn't set to `self.is_approved = auto_approve`? Feels weird to have 3 different behaviours?